### ISSUE TITLE
`member-delimiter-style`: value 'always' should be boolean when using config factory

### ISSUE THREAD

**Issue by attributeerror** [eslint-stylistic/eslint-stylistic#789]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@stylistic/eslint-plugin@4.2.0` for the project I'm working on.

I was getting the below error when trying to run ESLint against my project using the below configuration.  
Config:
```js
stylistic.configs.customize( {
	quotes: 'single',
	indent: 2,
	semi: 'always',
	arrowParens: 'always',
	braceStyle: '1tbs',
	blockSpacing: 'always',
	quoteProps: 'consistent',
	commaDangle: 'always',
} ),
```

Error:
```
Error - 1:10:20 PM] Error: Key "rules": Key "@stylistic/member-delimiter-style":
	Value "always" should be boolean.
```

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@stylistic/eslint-plugin/dist/configs.js b/node_modules/@stylistic/eslint-plugin/dist/configs.js
index 0065652..7663c14 100644
--- a/node_modules/@stylistic/eslint-plugin/dist/configs.js
+++ b/node_modules/@stylistic/eslint-plugin/dist/configs.js
@@ -255,14 +255,14 @@ function customize(options = {}) {
     "@stylistic/member-delimiter-style": ["error", {
       multiline: {
         delimiter: semi ? "semi" : "none",
-        requireLast: semi
+        requireLast: !!semi
       },
       multilineDetection: "brackets",
       overrides: {
         interface: {
           multiline: {
             delimiter: semi ? "semi" : "none",
-            requireLast: semi
+            requireLast: !!semi
           }
         }
       },
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

### Reproduction

https://stackblitz.com/edit/github-4mp68kss?file=eslint.config.js

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
The `semi` option in the factory function should be a boolean. See [here](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/packages/eslint-plugin/dts/options.d.ts#L27).
